Production/python/customiseHLT.py

```diff
@@ -0,0 +1,327 @@
+  process.genParticleTable.variables.mass.expr = cms.string('mass')
+  process.genParticleTable.variables.mass.doc = cms.string('mass')
+
+  return process
+
+
+def customise(process):
+  process.NANOAODSIMoutput = cms.OutputModule("NanoAODOutputModule",
+      compressionAlgorithm = cms.untracked.string('LZMA'),
+      compressionLevel = cms.untracked.int32(9),
+      dataset = cms.untracked.PSet(
+          dataTier = cms.untracked.string('NANOAODSIM'),
+          filterName = cms.untracked.string('')
+      ),
+      fileName = cms.untracked.string('file:nano.root'),
+      outputCommands  = cms.untracked.vstring(
+          'drop *',
+          'keep nanoaodFlatTable_*Table_*_*',
+      )
+  )
+  process.load('PhysicsTools.NanoAOD.nano_cff')
+  process.load('RecoJets.JetProducers.ak4GenJets_cfi')
+  from PhysicsTools.NanoAOD.nano_cff import nanoAOD_customizeMC
+  from RecoJets.JetProducers.ak4GenJets_cfi import ak4GenJets 
+  #call to customisation function nanoAOD_customizeMC imported from PhysicsTools.NanoAOD.nano_cff
+  process = nanoAOD_customizeMC(process)
+
+  process.nanoAOD_step = cms.Path(process.HLTBeginSequence \
+    + process.HLTL2TauTagNNSequence \
+    + process.HLTGlobalPFTauHPSSequence \
+    + process.HLTHPSDeepTauPFTauSequenceForVBFIsoTau \
+    + process.nanoSequenceMC)
+  process.NANOAODSIMoutput_step = cms.EndPath(process.NANOAODSIMoutput)
+  
+  from PhysicsTools.JetMCAlgos.HadronAndPartonSelector_cfi import selectedHadronsAndPartons
+  process.selectedHadronsAndPartons = selectedHadronsAndPartons.clone(src = cms.InputTag("genParticlesForJetsNoNu"))
+  # set ak4GenJets producer 
+  # process.ak4GenJetsNoNu = ak4GenJets.clone( src = "genParticlesForJetsNoNu")
+  process.ak4GenJetsNoNu = ak4GenJets.clone( src = "genParticlesForJetsNoNu")#, bHadrons= cms.InputTag("selectedHadronsAndPartons","bHadrons"), cHadrons= cms.InputTag("selectedHadronsAndPartons","cHadrons"),partons= cms.InputTag("selectedHadronsAndPartons","physicsPartons"), )
+  process.genJetFlavourInfos = ak4JetFlavourInfos.clone(  
+    jets = cms.InputTag( "ak4GenJetsNoNu" ),
+    bHadrons= cms.InputTag("selectedHadronsAndPartons","bHadrons"), 
+    cHadrons= cms.InputTag("selectedHadronsAndPartons","cHadrons"),
+    partons= cms.InputTag("selectedHadronsAndPartons","physicsPartons"), )
+  
+  process.genParticlesForJetsNoNu = cms.EDProducer("InputGenJetsParticleSelector",
+    src = cms.InputTag("finalGenParticles"),
+    ignoreParticleIDs = cms.vuint32(
+        #  1000022,
+        #  1000012, 1000014, 1000016,
+        #  2000012, 2000014, 2000016,
+        #  1000039, 5100039,
+        #  4000012, 4000014, 4000016,
+        #  9900012, 9900014, 9900016,
+        #  39, 
+         12,14,16 ), # ignore for neutrinos
+    partonicFinalState = cms.bool(False),
+    excludeResonances = cms.bool(False),
+    excludeFromResonancePids = cms.vuint32(12, 14, 16), 
+    tausAsJets = cms.bool(False)
+)
```

Comment:
```suggestion
  process.load('RecoJets.Configuration.GenParticles_cff')
```

Reply:
I am looking at https://github.com/cms-sw/cmssw/blob/master/RecoJets/Configuration/python/GenJetParticles_cff.py.

This should be GenJetParticles_cff not GenParticles_cff? 

In addition, I see in the official repo there is `excludeFromResonancePids = cms.vuint32(12, 13, 14, 16),` @ofivite suggested removing 13 IIRC. Do you know if 13 should be on the list?

Reply:
> This should be GenJetParticles_cff not GenParticles_cff?

Indeed, it should be `GenJetParticles_cff`



> In addition, I see in the official repo there is `excludeFromResonancePids = cms.vuint32(12, 13, 14, 16),` @ofivite suggested removing 13 IIRC. Do you know if 13 should be on the list?

In that config two producers are defined: `genParticlesForJetsNoNu` and `genParticlesForJetsNoMuNoNu`. Muons are excluded in the second one, while for our purposes `genParticlesForJetsNoNu` should be used.

Reply:
And `excludeResonances`is set to `False`, so what is specified in `excludeFromResonancePids` is not relevant.

Reply:
Ah ok, that answers my questions. Thanks!